### ISSUE TITLE
issue during inferecing the model of dvae.pth file

### ISSUE THREAD

**Issue by thenameisAnurag** [anhnh2002/XTTSv2-Finetuning-for-New-Languages#68]
i have trained the model  on the opensource 
https://commonvoice.mozilla.org/en/datasets
this data set and i have did the inferecing of the model 

<img width="1854" height="1080" alt="Image" src="https://github.com/user-attachments/assets/6b5613b5-0bd9-4190-b096-2b91f14cb998" />

import os
import torch
import torchaudio
from tqdm import tqdm
from underthesea import sent_tokenize

from TTS.tts.configs.xtts_config import XttsConfig
from TTS.tts.models.xtts import Xtts
from TTS.tts.models.xtts import XttsAudioConfig

# Allow safe globals for PyTorch >=2.6
torch.serialization.add_safe_globals([XttsConfig, XttsAudioConfig])

# Device
device = "cuda:0" if torch.cuda.is_available() else "cpu"

# === Paths ===
BASE_DIR = "/home/anuragmishra/Anurag/TTS/XTTS-v2"
xtts_config = os.path.join(BASE_DIR, "/home/anuragmishra/Anurag/TTS/XTTS-v2/config.json")
xtts_vocab = os.path.join(BASE_DIR, "/home/anuragmishra/Anurag/TTS/XTTS-v2/vocab.json")
xtts_checkpoint_dir = os.path.join(BASE_DIR, "/home/anuragmishra/Anurag/TTS/XTTS-v2/XTTSv2-Finetuning-for-New-Languages/checkpoints/XTTS_v2.0_original_model_files")
dvae_checkpoint = os.path.join(BASE_DIR, "/home/anuragmishra/Anurag/TTS/XTTS-v2/model.pth")
speaker_audio_file = os.path.join(BASE_DIR, "XTTSv2-Finetuning-for-New-Languages/cv-corpus-22.0-2025-06-20/hi/clips/common_voice_hi_23795238.wav")

# === Load XTTS model ===
config = XttsConfig()
config.load_json(xtts_config)
XTTS_MODEL = Xtts.init_from_config(config)
XTTS_MODEL.load_checkpoint(config, checkpoint_dir=xtts_checkpoint_dir, vocab_path=xtts_vocab, eval=True)

# Load fine-tuned DVAE into correct submodule
state_dict = torch.load(dvae_checkpoint, map_location=device, weights_only=False)
if hasattr(XTTS_MODEL, "dvae"):
    XTTS_MODEL.dvae.load_state_dict(state_dict, strict=False)
elif hasattr(XTTS_MODEL, "dvae_model"):
    XTTS_MODEL.dvae_model.load_state_dict(state_dict, strict=False)
else:
    raise AttributeError("❌ No DVAE module found in XTTS model.")
print("✅ Base XTTS + fine-tuned DVAE loaded!")

XTTS_MODEL.to(device).eval()

# === Inference ===
tts_text = "नमस्ते! आज का मौसम बहुत अच्छा है।"
lang = "hi"

gpt_cond_latent, speaker_embedding = XTTS_MODEL.get_conditioning_latents(
    audio_path=speaker_audio_file,
    gpt_cond_len=XTTS_MODEL.config.gpt_cond_len,
    max_ref_length=XTTS_MODEL.config.max_ref_len,
    sound_norm_refs=XTTS_MODEL.config.sound_norm_refs,
)

tts_texts = sent_tokenize(tts_text)
wav_chunks = []
for text in tqdm(tts_texts):
    out = XTTS_MODEL.inference(
        text=text,
        language=lang,
        gpt_cond_latent=gpt_cond_latent,
        speaker_embedding=speaker_embedding,
        temperature=0.7,
        top_p=0.8,
    )
    wav_chunks.append(torch.tensor(out["wav"]))

out_wav = torch.cat(wav_chunks, dim=0).unsqueeze(0).cpu()
torchaudio.save("output_inference.wav", out_wav, 24000)
print("🎙️ Saved to output_inference.wav")

<img width="1854" height="1080" alt="Image" src="https://github.com/user-attachments/assets/47e59772-4485-46e5-bb57-ee3d64453f9f" />

can any one guide me where i am going wrong
